python-avd/pyavd/_eos_designs/structured_config/mlag/__init__.py

```diff
@@ -377,6 +377,28 @@ def _router_bgp_mlag_peer_group(self) -> dict:
 
         router_bgp["peer_groups"] = [strip_empties_from_dict(peer_group)]
 
+        router_bgp["address_family_ipv4"] = {"peer_groups": []}
+        if self.shared_utils.use_separate_peer_group_for_mlag_vrfs:
+            router_bgp["peer_groups"].append(
+                {
+                    "name": self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.name,
+                    "type": "ipv4",
+                    "remote_as": self.shared_utils.bgp_as,
+                    "next_hop_self": True,
+                    "description": AvdStringFormatter().format(self.inputs.mlag_bgp_peer_group_description, mlag_peer=self.shared_utils.mlag_peer),
+                    "password": self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.password,
+                    "bfd": self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.bfd or None,
+                    "maximum_routes": 12000,
+                    "route_map_in": "RM-MLAG-PEER-IN",
+                },
+            )
+            if self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.structured_config:
+                self.custom_structured_configs.nested.router_bgp.peer_groups.obtain(self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.name)._deepmerge(
+                    self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.structured_config, list_merge=self.custom_structured_configs.list_merge_strategy
+                )
+            address_family_ipv4_vrf_peer_group = {"name": self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.name, "activate": True}
+            router_bgp["address_family_ipv4"]["peer_groups"].append(address_family_ipv4_vrf_peer_group)
+
         if self.shared_utils.underlay_ipv6:
```

Comment:
move this block closer to the undelray peer group to be clearer

Reply:
Done!